tapchannel/aux_traffic_shaper.go

```diff
@@ -308,11 +310,19 @@ func (s *AuxTrafficShaper) ProduceHtlcExtraData(totalAmount lnwire.MilliSatoshi,
 	)
 	numAssetUnits := numAssetUnitsFp.ScaleTo(0).ToUint64()
 
-	// We now know how many units we need. We take the asset ID from the
-	// RFQ so the recipient can match it back to the quote.
-	assetId, err := quote.Request.AssetSpecifier.UnwrapIdOrErr()
-	if err != nil {
-		return 0, nil, fmt.Errorf("quote has no asset ID: %w", err)
+	var assetId asset.ID
+
+	switch {
+	case quote.Request.AssetSpecifier.HasId():
+		assetId = *quote.Request.AssetSpecifier.UnwrapIdToPtr()
+
+	case quote.Request.AssetSpecifier.HasGroupPubKey():
+		// If a group key is defined in the quote we place the hash of
+		// the group key as the dummy asset ID in the HTLC. This asset
+		// balance in the HTLC is just a hint and the actual asset IDs
+		// will be picked later in the process.
+		groupKey := quote.Request.AssetSpecifier.UnwrapGroupKeyToPtr()
+		assetId = sha256.Sum256(groupKey.SerializeCompressed())
```

Comment:
Do we actually need to start using a specifier for the HTLC balance TLVs? 

Otherwise, we're introducing some overloading here, which is likely to be the source of confusion/bugs down the line. 

As an example, if we're using a fake asset ID (which is actually the group key hash), won't all the logic to create the allocations for th actual TAP tree in the HTLC/commitment outputs fail?